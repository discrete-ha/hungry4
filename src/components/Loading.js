import React from 'react';
import { connect } from 'react-redux';

const style = {
    loadingImage:{
        width: 50,
        height: 50,
        zIndex: 10,
        position: 'fixed',
        left: 0,
        right: 0,
        top: '30%',
        margin: 'auto'
    },
    loadingWrapper:{
        position: 'fixed',
        width: '100%',
        height: '100%',
        zIndex: 9,
        backgroundColor: 'rgba(0, 0, 0, 0.3)'
    },
    loadingMessage:{
        position: 'absolute',
        top: '30%',
        paddingTop: 50,
        width: '100%',
        height: 20,
        textAlign: 'center',
        color: '#fff',
        fontSize: 20
    }
    
}

export class Loading extends React.Component {

    render(){
        if (this.props.isFetching)
            return (
                <div style={style.loadingWrapper}>
                    <img alt='loading' style={style.loadingImage}src="./loader.gif"/>
                    <div style={style.loadingMessage}>{this.props.loadingMessage}</div>
                </div>
            );

        return null;
    }
}

let mapStateToProps = (state) => {
    return {
    	isFetching: state.isFetching,
        loadingMessage: state.loadingMessage
    };
}

export default connect(mapStateToProps)(Loading);
