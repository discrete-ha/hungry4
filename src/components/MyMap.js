import React from 'react';
import { withScriptjs, withGoogleMap, GoogleMap, Marker} from "react-google-maps";

const MyMap = withScriptjs(withGoogleMap(props =>{

  return <GoogleMap    
    defaultZoom={17}
    defaultCenter={props.center}
  >
    <Marker
      position={props.center}
    />
  </GoogleMap>
}
));

export default MyMap;
