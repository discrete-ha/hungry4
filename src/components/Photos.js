import React from 'react';
import { setImageGallery } from '../actions/place';
import { connect } from 'react-redux';
import ImageGallery from 'react-image-gallery';
import "react-image-gallery/styles/css/image-gallery.css";

const style = {
    thumnaiWrapper:{
        marginTop: 10,
        textAlign: 'center'
    },
    thumbs:{
        width:88,
        height:88,
        marginLeft: 10,
        cursor: 'pointer'
    },
    galleryScreen:{
        height: '100%',
        width: '100%',
        position: 'fixed',
        zIndex: 9,
        top:0,
        left:0,
        background: 'rgba(0,0,0,0.8)'
    },
    galleryWrapper:{
        width: '100%',
        maxWidth: 500,
        left: 0,
        right: 0,
        top: 0,
        paddingTop: '10%',
        margin: 'auto',
        position: 'fixed',
        zIndex: 10
    }
}

export class Photos extends React.Component {

    showImageGallery(index){
        this.props.setImageGallery(index);
    }

    renderImageGallery(){
        let showImageIndex = this.props.showImageIndex;
        if (showImageIndex !== null) {
            return (
                <div>
                    <div style={style.galleryScreen} onClick={this.showImageGallery.bind(this, null)}></div>
                    <div style={style.galleryWrapper}>
                        <ImageGallery startIndex={showImageIndex} items={this.props.photos}/>
                    </div>
                </div>);    
        }else{
            return null;
        }
    }

    renderPhoto(){
        return this.props.photos.map((photo, index)=>{
            return <img key={index} alt='place photos' src={photo.thumbnail} style={style.thumbs} 
            onClick={this.showImageGallery.bind(this, index)} />
        });
    }

    render(){
        if (this.props.photos) {
            return (
                <div style={style.thumnaiWrapper}>
                    {this.renderImageGallery()}
                    {this.renderPhoto()}
                </div>
            );    
        }
        return null;
    }
}


let mapStateToProps = (state) => {
    return {
        showImageIndex: state.showImageIndex
    };
}

let mapDispatchToProps = (dispatch) => {
    return {
        setImageGallery: (index) => dispatch(setImageGallery(index))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Photos);

