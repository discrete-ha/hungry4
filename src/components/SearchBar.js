import React from 'react';
import { connect } from 'react-redux';
import FaSearch from 'react-icons/lib/fa/search';
import FaSearchClear from 'react-icons/lib/fa/times-circle';
import { setSearchText, setPlaceList } from '../actions/place';
import { setFetching } from '../actions/system';

const style = {
	section:{
		position: 'relative',
		fontSize: 18,
		padding: '0 8px 8px 8px',
		// backgroundColor: "rgb(200, 0, 0)",
		boxSizing: "border-box",
	},
	searchWrapper:{
		position: 'relative',
		maxWidth: 500,
		width: '100%',
		margin: 'auto'
	},
	input:{
		borderRadius: 10,
		paddingLeft: 40,
		paddingRight: 40,
		width: '100%',
		height: 30,
		border:'none',
		boxSizing: 'border-box',
		outline: 'none',
		fontSize: 15
	},
	icon:{
		position: 'absolute',
		top: 3,
		color: '#828282'
	}
}

export class SearchBar extends React.Component {

	constructor(props){
       super(props);
       this.timeout = 0;
    }

	handleSearchText(event){

		let searchText = event.target.value.trim();
		if(this.timeout) 
			clearTimeout(this.timeout);

	    this.timeout = setTimeout( () => {

			if ( this.props.searchText !== searchText && searchText.length > 0 ){
				this.props.setSearchText(searchText);
			}else if(searchText.length === 0){
				this.props.setPlaceList([]);
			}

	    }, 500);

	}

	clearText(){
		this.refs.searchInput.value = '';
		this.props.setSearchText('');
		this.props.setPlaceList([]);
	}

	render(){
		return (<section style={style.section}>
					<div style={style.searchWrapper}>
						<label style={{...style.icon,left:15}}>
							<FaSearch />
						</label>
						<input placeholder="Search" style={style.input} ref="searchInput" onChange={this.handleSearchText.bind(this)}/>
						<a onClick={this.clearText.bind(this)} style={{...style.icon, right:15, cursor: 'pointer'}}><FaSearchClear /></a>
					</div>
				</section>);
	}
}

let mapStateToProps = (state) => {
	return {
		searchText: state.searchText,
		foursquare: state.foursquare
	};
}

let mapDispatchToProps = (dispatch) => {
	return {
		setSearchText: (text) => dispatch(setSearchText(text)),
		setFetching: (flag) => dispatch(setFetching(flag)),
		setPlaceList: (data) => dispatch(setPlaceList(data))
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(SearchBar);
