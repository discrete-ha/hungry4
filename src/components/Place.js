import React from 'react';
import { connect } from 'react-redux';
import { setDetailPlace, setShowingPlace, setRandomPlace } from '../actions/place';
import { setFetching } from '../actions/system';
import MyMap from './MyMap';
import Foursquare from '../services/Foursquare';
import Photos from './Photos';
import { rubberBand , fadeIn} from 'react-animations'
import Radium, {StyleRoot} from 'radium';
import ArrowCircleRight from 'react-icons/lib/fa/arrow-circle-right';

let RANDOM_ITEM_ID = 'randomitem';

const style = {
	placeWrapper:{
		margin: 'auto',
		maxWidth: 600,
		width: '100%',
		border: 'none',
	    borderRadius: 10,
	    backgroundColor: 'rgba(255, 255, 255, 0.2)',
	    padding: 3,
	    marginBottom: 10
	},
	title:{
		padding: 3,
		fontSize: 20,
		fontWeight: 500,
		color: '#fff'
	},
	childText:{
		padding: 3,
		fontSize: 14,
		fontWeight: 500,
		color: 'rgba(255, 255, 255, 1)'
	},
	link:{
		textDecoration: 'underline',
    	color: '#c5e2ff'
	},
	mapWrapper:{
		maxWidth:600,
		boxSizing: "border-box",
		width:'100%',
		height:320,
		padding:10,
		margin: 'auto'
	},
	pin:{
		color: 'rgb(200, 50, 50)',
		fontSize: 20
	},
    img:{
        width: 60,
        height: 60,
        position: 'absolute',
	    left: 0,
	    right: 0,
	    top: 0,
	    bottom: 0,
	    margin: 'auto',
	    animation: 'x 2s',
		animationName: Radium.keyframes(rubberBand, 'rubberBand'),
		animationIterationCount:'infinite'
    },
    randomWrapper:{
    	cursor: 'pointer', 
    	height: 100, 
    	position:'relative'
    },
    placeDetail:{
    	animation: 'x 1s',
		animationName: Radium.keyframes(fadeIn, 'fadeIn'),
    },
    nextWapper:{
    	position: 'relative'
    },
    nextIcon:{
    	position: 'absolute',
    	top:10,
    	right:10,
    	color: '#fff',
    	fontSize: 30,
    	cursor: 'pointer'
    }
}

export class Place extends React.Component {

    constructor(props){
       super(props);
       this.showRandomPlace = this.showRandomPlace.bind(this);
       this.showDetail = this.showDetail.bind(this);
    }

	showRandomPlace(){
		let places = this.props.randomList;
		let randIndex = Math.floor( (Math.random() * (places.length - 1) ) );
		let randomPlace = places[randIndex];
		this.props.setRandomPlace(randomPlace);
		this.props.setShowingPlace(RANDOM_ITEM_ID);
		if (!this.props.placeDetail[randomPlace.id]) {
			this.getPlaceDetail(randomPlace.id);
		}
	}

	getPlaceDetail(placeId){
		this.props.setFetching(true);
	    Foursquare.getPlaceDetail(placeId, (venue)=>{
			this.props.setDetailPlace(venue);
	        this.props.setFetching(false);
		});
	}

	showDetail(placeId){
		if (placeId === this.props.showingPlace) {
			this.props.setShowingPlace(null);
		}else{
			this.props.setShowingPlace(placeId);
			if (!this.props.placeDetail[placeId]) {
				this.getPlaceDetail(placeId);
			}
		}
	}

	showNextRandom(){
		return (<div style={style.nextWapper}>
				<ArrowCircleRight style={style.nextIcon}  onClick={this.showRandomPlace}/>
			</div>);
	}

    render(){
        let link = null;
        let place = this.props.isRandom ? this.props.randomPlace : this.props.place;

        if (this.props.isRandom && this.props.showingPlace !==  RANDOM_ITEM_ID) {
        	return <div style={{...style.placeWrapper, ...style.randomWrapper}} onClick={this.showRandomPlace}>
        			<StyleRoot>
						<img style={style.img} src="./random.png" />
					</StyleRoot>
				</div>;
        }else{
        	let placeDetail = this.props.placeDetail;
			let placeDetailElement = null; 
			if (place.url) {
				link = <div style={style.childText}><a href={place.url} target="blank" style={style.link} >{place.url}</a></div>;
			}

			if ( placeDetail[place.id] && ( this.props.showingPlace === place.id || this.props.showingPlace ===  RANDOM_ITEM_ID ) ) {
				placeDetailElement = (
					<StyleRoot>
						<div style={style.placeDetail}>
							<Photos photos={placeDetail[place.id].photos} />
							<div style={style.mapWrapper}>
								<MyMap
									center={{lat:placeDetail[place.id].location.lat, lng:placeDetail[place.id].location.lng}}
									googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyCgFwkfQzeKxUQNlMuooiFWt6nLJovy7bw&v=3.exp&libraries=places"
									loadingElement={<div style={{ height: `300px` }} />}
								    containerElement={<div style={{ height: `300px` }} />}
								    mapElement={<div style={{ height: `300px` }} />}
							    />
						    </div>
						</div>
					</StyleRoot>
				);
			}

		    return (<div style={style.placeWrapper}>
		    		{this.showNextRandom()}
			    	<div style={{cursor: 'pointer'}} onClick={()=>this.showDetail(place.id)}>
				    	<div style={style.title}>{place.name}</div>
				    	<div style={style.childText}>{place.location.formattedAddress[0]} {place.location.formattedAddress[1]}</div>
				    	<div style={style.childText}>{place.location.distance}m</div>
			    	</div>
			    	{link}
			    	<div style={style.childText}>{place.contact.formattedPhone}</div>
			    	{placeDetailElement}
			    </div>);
        }
    }
}

let mapStateToProps = (state) => {
    return {
    	foursquare :state.foursquare,
    	placeDetail: state.placeDetail,
    	showingPlace: state.showingPlace,
    	randomList: state.randomList,
    	randomPlace: state.randomPlace
    };
}

let mapDispatchToProps = (dispatch) => {
    return {
    	setFetching: (flag) => dispatch(setFetching(flag)),
    	setDetailPlace: (detail) => dispatch(setDetailPlace(detail)),
    	setShowingPlace: (placeId) => dispatch(setShowingPlace(placeId)),
    	setRandomPlace: (data) => dispatch(setRandomPlace(data))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Place);
