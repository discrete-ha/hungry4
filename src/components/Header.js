import React from 'react';
import SearchBar from './SearchBar';

const style = {
    header:{
        // backgroundColor: "rgb(200, 0, 0)",
        padding: 10,
        color: "#fff",
        textAlign: "center"
    },
    appTitle:{
        fontSize: 25,
        fontWeight: 900
    },
    text:{
        fontSize:14
    },
    link:{
        textDecoration: 'none',
        color: '#fff'
    }
}

class Header extends React.Component {

    render(){
        return (<div style={style.header}>
          <div style={style.text}>what are you </div>
          <div style={style.appTitle}><a style={style.link} href="/">HUNGRY4</a></div>
        </div>)
    }
}

export default Header;
