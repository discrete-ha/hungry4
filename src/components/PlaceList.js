import React from 'react';
import { connect } from 'react-redux';
import Place from './Place';
import { bounceInLeft } from 'react-animations'
import Radium, {StyleRoot} from 'radium';

const style = {
	wrapper:{
		// background: "linear-gradient(217deg, rgba(51, 0, 0, 0.8), rgba(255, 0, 0, 0) 70.71%), linear-gradient(127deg, rgba(255, 82, 0, 0.73), rgba(255, 0, 0, 0) 70.71%), linear-gradient(336deg, rgba(99, 37, 48, 0.8), rgba(255, 0, 0, 0) 70.71%)",
		padding:10,
		flex: 1
	},
	place:{
		animation: 'x 1s',
		animationName: Radium.keyframes(bounceInLeft, 'bounceInLeft')
	}
}

export class PlaceList extends React.Component {

    render(){
    	if (this.props.placeList.length > 0) {

			let placeListElement = this.props.placeList.map(function(place, index){
				let interval = index * 0.2 > 1 ? 1 : index * 0.2;
				return (<StyleRoot key={place.id}>
					<div style={{...style.place, animation: interval+'s'}}>
						<Place place={place} isRandom={false}/>
					</div>
				</StyleRoot>)
			});

			return (<div style={style.wrapper}>
				{placeListElement}
			</div>);
    	}else if(this.props.randomList.length > 0){
    		return (<StyleRoot style={style.wrapper}>
    			<div style={ {...style.place, padding:10 } }>
					<Place isRandom={true} />
				</div>
			</StyleRoot>);
    	}else{
    		return null;
    	}
    }
}

let mapStateToProps = (state) => {
    return {
    	placeList: state.placeList,
    	randomList: state.randomList
    };
}

export default connect(mapStateToProps)(PlaceList);
