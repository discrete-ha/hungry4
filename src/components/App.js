import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setFetching } from '../actions/system';
import Loading from './Loading';
import Header from './Header';
import PlaceList from './PlaceList';
import SearchBar from './SearchBar';
import Foursquare from '../services/Foursquare';
import { setPlaceList, setRandomList } from '../actions/place';

const style = {
  appWrapper:{
    display: 'flex',
    flexDirection: 'column',
    minHeight: '100vh'
  }
}

class App extends Component {

  searchPlace(searchText){
    if (searchText.length === 0) 
      return false;

    this.props.setFetching(true);
    Foursquare.searchPlace(searchText, (places)=>{
      this.props.setPlaceList(places);
      this.props.setFetching(false);
    });
  }

  getRandomPlace(){
    Foursquare.getRandomPlace((places)=>{
      this.props.setRandomList(places);
      this.props.setFetching(false);
    });
  }

  componentWillMount(){
    this.getRandomPlace();
  }

  shouldComponentUpdate(nextProps){

    if (this.props.searchText !== nextProps.searchText) {
      this.searchPlace(nextProps.searchText);
    }
    return true;
  }

  render() {
    console.log('app render');
    return (
      <div style={style.appWrapper}>
        <Loading />
        <Header />
        <SearchBar />
        <PlaceList />
      </div>
    );
  }
}

let mapStateToProps = (state) => {
    return {
      isFetching: state.isFetching,
      foursquare: state.foursquare,
      searchText: state.searchText
    };
}

let mapDispatchToProps = (dispatch) => {
    return {
        setFetching: (flag) => dispatch(setFetching(flag)),
        setPlaceList: (data) => dispatch(setPlaceList(data)),
        setRandomList: (data) => dispatch(setRandomList(data))
    }
}

App = connect(mapStateToProps, mapDispatchToProps)(App);

export default App;
