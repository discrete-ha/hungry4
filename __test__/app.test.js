import React from 'react';
import renderer from 'react-test-renderer';
import testData from './test.data.json';
import Header from '../src/components/Header';
import { Loading } from '../src/components/Loading';
import MyMap from '../src/components/MyMap';
import { Place } from '../src/components/Place';
import { SearchBar } from '../src/components/SearchBar';

let defaultProps = {
	isFetching: true,
    searchText: '',
    foursquare:{
    	"ll": "35.648795,139.702237",
  		"radius": 1000
    },
    placeList:[],
    randomList:[],
    placeDetail:{},
    randomPlace:null,
    showingPlace:null
}

describe('app components render', () => {
  	test('Header', ()=>{
		const component = renderer.create(
			<Header />,
		);
		let headerSnapshot = component.toJSON();
		expect(headerSnapshot).toMatchSnapshot();
	});

  	test('Loading - fetching', ()=>{
		const component = renderer.create(
			<Loading isFetching={true}/>,
		);
		let loadingSnapshot = component.toJSON();
		expect(loadingSnapshot).toMatchSnapshot();
	});

	test('Loading - not fetching', ()=>{
		const component = renderer.create(
			<Loading />,
		);
		let loadingSnapshot = component.toJSON();
		expect(loadingSnapshot).toMatchSnapshot();
	});

	test('MyMap', ()=>{
		const component = renderer.create(
			<MyMap
				center={{lat:35.648795, lng:139.702237}}
				googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyCgFwkfQzeKxUQNlMuooiFWt6nLJovy7bw&v=3.exp&libraries=places"
				loadingElement={<div style={{ height: `300px` }} />}
			    containerElement={<div style={{ height: `300px` }} />}
			    mapElement={<div style={{ height: `300px` }} />}
		    />
		);
		let myMapSnapshot = component.toJSON();
		expect(myMapSnapshot).toMatchSnapshot();
	});

	test('Place - Random', ()=>{
		const component = renderer.create(
			<Place isRandom={true} />,
		);
		let placeSnapshot = component.toJSON();
		expect(placeSnapshot).toMatchSnapshot();
	});

	test('Place - Not random', ()=>{
		const component = renderer.create(
			<Place isRandom={false} 
				place={testData.place}
				{...defaultProps} />,
		);
		let placeSnapshot = component.toJSON();
		expect(placeSnapshot).toMatchSnapshot();
	});

	test('SearchBar', ()=>{
		const component = renderer.create(
			<SearchBar />,
		);
		let searchBarSnapshot = component.toJSON();
		expect(searchBarSnapshot).toMatchSnapshot();
	});

});


