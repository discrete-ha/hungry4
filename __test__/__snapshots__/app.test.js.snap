// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app components render Header 1`] = `
<div
  style={
    Object {
      "color": "#fff",
      "padding": 10,
      "textAlign": "center",
    }
  }
>
  <div
    style={
      Object {
        "fontSize": 14,
      }
    }
  >
    what are you 
  </div>
  <div
    style={
      Object {
        "fontSize": 25,
        "fontWeight": 900,
      }
    }
  >
    <a
      href="/"
      style={
        Object {
          "color": "#fff",
          "textDecoration": "none",
        }
      }
    >
      HUNGRY4
    </a>
  </div>
</div>
`;

exports[`app components render Loading - fetching 1`] = `
<div
  style={
    Object {
      "backgroundColor": "rgba(0, 0, 0, 0.3)",
      "height": "100%",
      "position": "fixed",
      "width": "100%",
      "zIndex": 9,
    }
  }
>
  <img
    alt="loading"
    src="./loader.gif"
    style={
      Object {
        "height": 50,
        "left": 0,
        "margin": "auto",
        "position": "fixed",
        "right": 0,
        "top": "30%",
        "width": 50,
        "zIndex": 10,
      }
    }
  />
</div>
`;

exports[`app components render Loading - not fetching 1`] = `null`;

exports[`app components render MyMap 1`] = `
<div
  style={
    Object {
      "height": "300px",
    }
  }
/>
`;

exports[`app components render Place - Not random 1`] = `
<div
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 0.2)",
      "border": "none",
      "borderRadius": 10,
      "margin": "auto",
      "marginBottom": 10,
      "maxWidth": 600,
      "padding": 3,
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <svg
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      preserveAspectRatio="xMidYMid meet"
      style={
        Object {
          "color": "#fff",
          "cursor": "pointer",
          "fontSize": 30,
          "position": "absolute",
          "right": 10,
          "top": 10,
          "verticalAlign": "middle",
        }
      }
      viewBox="0 0 40 40"
      width="1em"
    >
      <g>
        <path
          d="m31.7 20q0-0.6-0.4-1l-10.1-10.1q-0.4-0.4-1-0.4t-1 0.4l-2.1 2q-0.4 0.4-0.4 1t0.4 1l4.3 4.2h-11.3q-0.5 0-1 0.5t-0.4 1v2.8q0 0.6 0.4 1t1 0.5h11.3l-4.3 4.2q-0.4 0.4-0.4 1t0.4 1l2.1 2q0.4 0.4 1 0.4t1-0.4l10.1-10.1q0.4-0.4 0.4-1z m5.6 0q0 4.7-2.3 8.6t-6.3 6.2-8.6 2.3-8.6-2.3-6.2-6.2-2.3-8.6 2.3-8.6 6.2-6.2 8.6-2.3 8.6 2.3 6.3 6.2 2.3 8.6z"
        />
      </g>
    </svg>
  </div>
  <div
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <div
      style={
        Object {
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": 500,
          "padding": 3,
        }
      }
    >
      Haco cafe 匣
    </div>
    <div
      style={
        Object {
          "color": "rgba(255, 255, 255, 1)",
          "fontSize": 14,
          "fontWeight": 500,
          "padding": 3,
        }
      }
    >
      代官山町19-1 (白磁館 2A)
       
      Shibuya, Tokyo
    </div>
    <div
      style={
        Object {
          "color": "rgba(255, 255, 255, 1)",
          "fontSize": 14,
          "fontWeight": 500,
          "padding": 3,
        }
      }
    >
      130
      m
    </div>
  </div>
  <div
    style={
      Object {
        "color": "rgba(255, 255, 255, 1)",
        "fontSize": 14,
        "fontWeight": 500,
        "padding": 3,
      }
    }
  >
    <a
      href="http://ameblo.jp/haco--cafe/"
      style={
        Object {
          "color": "#c5e2ff",
          "textDecoration": "underline",
        }
      }
      target="blank"
    >
      http://ameblo.jp/haco--cafe/
    </a>
  </div>
  <div
    style={
      Object {
        "color": "rgba(255, 255, 255, 1)",
        "fontSize": 14,
        "fontWeight": 500,
        "padding": 3,
      }
    }
  >
    +81 3-5428-5035
  </div>
</div>
`;

exports[`app components render Place - Random 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 0.2)",
      "border": "none",
      "borderRadius": 10,
      "cursor": "pointer",
      "height": 100,
      "margin": "auto",
      "marginBottom": 10,
      "maxWidth": 600,
      "padding": 3,
      "position": "relative",
      "width": "100%",
    }
  }
>
  <div
    data-radium={true}
  >
    <img
      data-radium={true}
      src="./random.png"
      style={
        Object {
          "animation": "x 2s",
          "animationIterationCount": "infinite",
          "animationName": "rubberBand-radium-animation-8ac73554",
          "bottom": 0,
          "height": 60,
          "left": 0,
          "margin": "auto",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 60,
        }
      }
    />
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "@keyframes rubberBand-radium-animation-8ac73554 {
from{transform: scale3d(1, 1, 1);}
30%{transform: scale3d(1.25, 0.75, 1);}
40%{transform: scale3d(0.75, 1.25, 1);}
50%{transform: scale3d(1.15, 0.85, 1);}
65%{transform: scale3d(0.95, 1.05, 1);}
75%{transform: scale3d(1.05, 0.95, 1);}
to{transform: scale3d(1, 1, 1);}
}
",
        }
      }
    />
  </div>
</div>
`;

exports[`app components render SearchBar 1`] = `
<section
  style={
    Object {
      "boxSizing": "border-box",
      "fontSize": 18,
      "padding": "0 8px 8px 8px",
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "margin": "auto",
        "maxWidth": 500,
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <label
      style={
        Object {
          "color": "#828282",
          "left": 15,
          "position": "absolute",
          "top": 3,
        }
      }
    >
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m27.2 18.6q0-4.2-2.9-7.1t-7.1-2.9-7 2.9-3 7.1 2.9 7 7.1 3 7.1-3 2.9-7z m11.4 18.5q0 1.2-0.8 2.1t-2 0.8q-1.2 0-2-0.8l-7.7-7.7q-4 2.8-8.9 2.8-3.2 0-6.1-1.3t-5-3.3-3.4-5-1.2-6.1 1.2-6.1 3.4-5.1 5-3.3 6.1-1.2 6.1 1.2 5 3.3 3.4 5.1 1.2 6.1q0 4.9-2.7 8.9l7.6 7.6q0.8 0.9 0.8 2z"
          />
        </g>
      </svg>
    </label>
    <input
      onChange={[Function]}
      placeholder="Search"
      style={
        Object {
          "border": "none",
          "borderRadius": 10,
          "boxSizing": "border-box",
          "fontSize": 15,
          "height": 30,
          "outline": "none",
          "paddingLeft": 40,
          "paddingRight": 40,
          "width": "100%",
        }
      }
    />
  </div>
</section>
`;
